docs/source/what_is_jupyter.md

```diff
@@ -0,0 +1,412 @@
+- Process astronomical data from the James Webb Space Telescope (JWST) (CHECK/LINK)
+- Generate an image of a black hole by processing telescope data (CHECK/LINK)
+- Calculate the presence of gravitational waves from observatory data (CHECK/LINK)
+
+Those last three are real world examples that demonstrate the scientific
+community's usage of computational notebooks (CHECK). Because scientists,
+engineers, and other technical people so often need to perform and communicate
+sophisticated calculations to describe their work, these "computational
+notebooks" became a very popular way to share their work and ideas.
+
+Notebooks are not just for Nasa scientists either: Students, hobbyists, and
+business people commonly use the fast, interactive workflows in JupyterLab to
+make notebooks that help solve everyday problems. You can use them to explain
+your own ideas to others, to learn, to automate tasks at home or at work,
+visualize complex information, and more.
+
+When someone uses the term "notebook", they might be referring to:
+
+- A notebook file on their computer
+- The idea of combining computer code, explanatory text, images and more into
+  the "notebook format"
+- The "Jupyter Notebook" application, used to author and edit digital notebook
+  files
+- Jupyter's ``.ipynb`` notebook file format (used to save your notebook files
+  on your computer), which is interpreted by the ``nbformat`` software library
+
+And the term "Jupyter" might refer to:
+
+- "Project Jupyter", the overarching umbrella project
+- The "Jupyter Notebook" or "JupyterLab" editing programs (or other Jupyter
+  products)
+
+The name Jupyter comes from the three programming languages the project
+originally supported: Julia (ju), Python (pyt) and R (r).
+
+# How do the Jupyter notebook-editor programs work?
+
+It is common for notebook-editor programs like JupyterLab, or Jupyter
+Notebook, to share some features and workflows, because they are influenced
+by a common set of ideas about computational notebooks, their advantages,
+and how best to work with them effectively.
+
+Let's break down some of those ideas.
+
+## Interactive programming (the REPL)
+
+In the past, writing programs, running them, and seeing results was commonly
+a slower and more deliberative process than it is today.
+
+As time passed, new techniques were discovered that sped up the process between
+writing code, running it, and seeing results. Languages like Python were also
+later introduced that offered some unique advantages over older languages.
+
+For instance, Python is an interpreted language, so it does not need to be
+compiled (LINK) before it can be run (saving programmers steps and time).
+It is also a dynamically-typed language that does not require the programmer
+to specify the type of their data ahead of time before using it, which can
+sometimes save time and reduce the complexity of code (particularly for
+smaller, simpler programs).
+
```

Comment:
I think that this introduction is less about Jupyter and more about Python. Probably OK to delete, to get right to the point about the REPL, and specifically to say what Jupyter does with the REPL.